orchestrator/careplancontributor/service.go

```diff
@@ -259,7 +244,43 @@ func (s Service) withSessionOrBearerToken(next func(response http.ResponseWriter
 	}
 }
 
-func (s Service) handleNotification(_ any) error {
-	log.Info().Msg("TODO: Handle received notification")
+func (s Service) handleNotification(resource any) error {
+	notification := resource.(*coolfhir.SubscriptionNotification)
+	if notification == nil {
+		return &coolfhir.ErrorWithCode{
+			Message:    "failed to cast resource to notification",
+			StatusCode: http.StatusInternalServerError,
+		}
+	}
+
+	focusReference, err := notification.GetFocus()
+	if err != nil {
+		return err
+	}
+	if focusReference == nil {
+		return &coolfhir.ErrorWithCode{
+			Message:    "Notification focus not found",
+			StatusCode: http.StatusUnprocessableEntity,
+		}
+	}
+
```

Comment:
maybe some debug/info logging concerning that we have received a notification?

Reply:
As discussed, I will add improved logging around message delivery in general